### PR TITLE
DDF-2852 [2.10.x] Add functionality for retrieving a logged-in user's security claims

### PR DESCRIPTION
**Pull request by coyotesqrl**
#### What does this PR do?
Adds utility functionality to extract claims from a Subject
Also makes a small, unrelated docs change.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@AzGoalie @garrettfreibott @ricklarsen 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Build](https://github.com/orgs/codice/teams/build)
[Docs](https://github.com/orgs/codice/teams/docs)
[UI](https://github.com/orgs/codice/teams/ui)
[Security](https://github.com/orgs/codice/teams/security)
[Continuous Integration](https://github.com/orgs/codice/teams/continuous-integration)
[Solr](https://github.com/orgs/codice/teams/solr)
[IO](https://github.com/orgs/codice/teams/IO)
[OGC](https://github.com/orgs/codice/teams/OGC)
[Data](https://github.com/orgs/codice/teams/data)
[Core APIs](https://github.com/orgs/codice/teams/core-apis)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@stustison

#### How should this be tested? (List steps with links to updated documentation)
CI build sufficient.

#### Any background context you want to provide?
N/A

#### What are the relevant tickets?
[DDF-2852](https://codice.atlassian.net/browse/DDF-2852)

#### Checklist:
- [X] Documentation Updated
- [ ] Change Log Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: platform/security/platform-security-core-api/pom.xml [hunk 1]
```diff
@@ -195,17 +195,17 @@
                                         <limit>
                                             <counter>INSTRUCTION</counter>
                                             <value>COVEREDRATIO</value>
-                                            <minimum>0.55</minimum>
```
Comment (coyotesqrl) [inline platform/security/platform-security-core-api/pom.xml]: General comment on the changes to these values: I err on the side of caution with the floating point calculations jacoco makes and set these .02 below the coverage values I saw in my environment. The new code was almost completely covered; there is one short branch that does not currently have a test.

File: distribution/docs/src/main/resources/_contents/_using/using-catalog-search-ui-contents.adoc [hunk 2]
```diff
@@ -284,6 +284,17 @@ image::catalog-ui-settings-options.png[]
 * Visualization: Select options for viewing data collections
 * Hidden: View or edit a list of records that have been hidden from search results.
 
+==== ${catalog-ui} User Profile
+
+When logged in, you can view details about your account using the *User Profile* (image:user-icon.png[user]) menu on the Navigation Bar.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_using/using-catalog-search-ui-contents.adoc]: We are avoiding "you" in the documentation. Maybe "When logged in, the user can..."

File: catalog/ui/catalog-ui-search/src/main/webapp/component/user/user.less [hunk 3]
```diff
@@ -26,5 +26,16 @@
         text-align: center;
     }
 
-
+    .claims-table {
+        text-align: left;
+        font-size: 1rem;
+        table-layout: fixed;
+        > tbody {
```
Comment (djblue) [inline catalog/ui/catalog-ui-search/src/main/webapp/component/user/user.less]: This selector could be reduce to `tbody th { ... }`.

**Comment by coyotesqrl** [comment]
This PR has been altered based on conversations with @pklinef and @andrewkfiedler regarding future functionality.

**Pull request merged**